src/main/java/com/project/bookstore/service/BookServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.project.bookstore.service;
+
+import com.project.bookstore.mapper.BookMapper;
+import com.project.bookstore.model.Book;
+import com.project.bookstore.model.dto.BookDto;
+import com.project.bookstore.model.dto.CreateBookRequestDto;
+import com.project.bookstore.model.dto.UpdateBookRequestDto;
+import com.project.bookstore.repository.BookRepository;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
+        this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+    }
+
+    @Override
+    public BookDto save(CreateBookRequestDto bookDto) {
+        Book book = bookMapper.toEntity(bookDto);
+        Book createdBook = bookRepository.save(book);
+        return bookMapper.toDto(createdBook);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        List<Book> books = bookRepository.findByIsDeletedFalse();
+        return books.stream()
+                .map(bookMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
+        if (book.isDeleted()) {
+            throw new EntityNotFoundException("Book not found with id: " + id);
+        }
+        return bookMapper.toDto(book);
```

Comment:
inline using Optional#map
```suggestion
       return bookRepository.findById(id)
        .map(bookMapper::toDto)
                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
        if (book.isDeleted()) {
            throw new EntityNotFoundException("Book not found with id: " + id);
        }
```